libs/prelude/Prelude/EqOrd.idr

```diff
@@ -136,9 +136,13 @@ Ord Int where
   (>) x y = intToBool (prim__gt_Int x y)
   (>=) x y = intToBool (prim__gte_Int x y)
 
+public export
+compareInteger : (x : Integer) -> (y : Integer) -> Ordering
+compareInteger x y = if (intToBool $ prim__lt_Integer x y) then LT else if (intToBool $ prim__eq_Integer x y) then EQ else GT
```

Comment:
Why are you not reusing `if x < y then LT else if x == y then EQ else GT`?

Reply:
Because `compareInteger` is defined before `OrdInteger`.

If I try to use `if x < y then LT else if x == y then EQ else GT`, I get:
```
 `-- Prelude.EqOrd line 141 col 24:
     While processing right hand side of compareInteger. Can't find an implementation for Ord Integer.
     
     Prelude.EqOrd:141:25--141:30
      137 |   (>=) x y = intToBool (prim__gte_Int x y)
      138 | 
      139 | public export
      140 | compareInteger : (x : Integer) -> (y : Integer) -> Ordering
      141 | compareInteger x y = if x < y then LT else if x == y then EQ else GT
                                    ^^^^^
```

Reply:
You could insert it after the `Ord` declaration and have `compareInteger = compare` though